website/docs/r/iam_role.html.markdown

```diff
@@ -91,6 +65,164 @@ resource "aws_iam_role" "instance" {
 }
 ```
 
+### Example of Exclusive Inline Policies
+
+This example creates an IAM role with two inline IAM policies. If someone adds another inline policy out-of-band, on the next apply, Terraform will remove that policy. If someone deletes these policies out-of-band, Terraform will recreate them.
+
+```hcl
+resource "aws_iam_role" "example" {
+  name               = "yak_role"
+  assume_role_policy = data.aws_iam_policy_document.instance_assume_role_policy.json # (not shown)
+
+  inline_policy {
+    name = "my_inline_policy"
+
+    policy = <<EOF
+{
+  "Statement": [
+    {
+      "Action": [
+        "ec2:Describe*"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ],
+  "Version": "2012-10-17"
+}
+EOF
+  }
+
+  inline_policy {
+    name   = "policy-8675309"
+    policy = data.aws_iam_policy_document.inline_policy.json
+  }
+}
+
+data "aws_iam_policy_document" "inline_policy" {
+  statement {
+    actions   = ["ec2:DescribeAccountAttributes"]
+    resources = ["*"]
+  }
+}
+```
+
+### Example of Removing Inline Policies
+
+This example creates an IAM role with what appears to be empty IAM `inline_policy` argument instead of using `inline_policy` as a configuration block. The result is that if someone were to add an inline policy out-of-band, on the next apply, Terraform will remove that policy.
+
+
+```hcl
+resource "aws_iam_role" "example" {
+  name               = "yak_role"
+  assume_role_policy = data.aws_iam_policy_document.instance_assume_role_policy.json # (not shown)
+
+  inline_policy {}
+}
+```
+
+### Example of Exclusive Managed Policies
+
+This example creates an IAM role and attaches two managed IAM policies. If someone attaches another managed policy out-of-band, on the next apply, Terraform will detach that policy. If someone detaches these policies out-of-band, Terraform will attach them again.
+
+```hcl
+resource "aws_iam_role" "example" {
+  name                = "yak_role"
+  assume_role_policy  = data.aws_iam_policy_document.instance_assume_role_policy.json # (not shown)
+  managed_policy_arns = [aws_iam_policy.policy_one.arn, aws_iam_policy.policy_two.arn]
+}
+
+resource "aws_iam_policy" "policy_one" {
+  name = "policy-618033"
+
+  policy = <<EOF
```

Comment:
Nit: Similarly here and below with `jsonencode()`/`aws_iam_policy_document`

Reply:
done